### ISSUE TITLE
Attempting to extend a declared class should give a better error message

### ISSUE THREAD

**Issue by georgewfraser** [microsoft/TypeScript#171]
From the old issue tracker:

http://typescript.codeplex.com/workitem/2485

``` typescript
declare module m.SuperClass {
    function moduleFunction();
}

declare module m {
    class SuperClass {
        constructor();
    }
}

declare module m {
    class SubClass extends m.SuperClass {
        constructor();
    }
}
```

Gives the baffling error message: 

```
error TS2186: Type name 'm.SuperClass' in extends clause does not reference constructor function for 'SuperClass'.
```
